cmd/scimd.go

```diff
@@ -41,6 +49,32 @@ Complete documentation is available at ...`,
 			fmt.Fprintln(os.Stderr, err)
 			os.Exit(1)
 		}
+
+		// try to connect to storage, if not db up: notify error and os.Exit(1)
+		// (todo) manage storage switching
+		if err := dbConnect(); err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+
+		// Get unique attributes from loaded schemas
+		keys, err := attr.GetUniqueAttributes()
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+
+		// when db exists and it is up and running: get collection + ensure index SYNC (we need it to be blocking)
+		// Note that the session executing EnsureIndex will be blocked for as long as it takes for the index to be built.
+		// So I think the next line will be blocking
```

Comment:
@alelb Do you think or... are you sure?

Reply:
I'm pretty sure :)..once the indexes are ensured, pre-run execution is continued.